src/app/child-dev-project/children/model/child.ts

```diff
@@ -34,43 +34,23 @@ export class Child extends Entity {
 
   @DatabaseField() name: string;
   @DatabaseField() projectNumber: string; // project number
-  @DatabaseField({ dataType: "string" }) gender: Gender; // M or F
   @DatabaseField({ dataType: "date-only" }) dateOfBirth: Date;
   @DatabaseField() motherTongue: string = "";
+  @DatabaseField({ dataType: "string" }) gender: Gender; // M or F
   @DatabaseField() religion: string = "";
 
   @DatabaseField() center: string = "";
   @DatabaseField() admissionDate: Date;
   @DatabaseField() status: string = "";
 
-  /** current school (as determined through the ChildSchoolRelation docs) set during loading through ChildrenService */
-  schoolId: string = "";
-  /** current class (as determined through the ChildSchoolRelation docs) set during loading through ChildrenService */
-  schoolClass: string = "";
-
-  @DatabaseField() address: string = "";
```

Comment:
We need to make really sure that we are not _ever_ missing an entity field in the config, otherwise people would just delete properties from objects in the database whenever they save them!
Can we think of any safeguards? Or should we consider switching the approach to one where fields not mentioned in the schema are simply saved without transformation as well?

Reply:
In that case, I would be worried about saving too much with the entity, if we just save everything. But what about the revisions? Aren't older revisions still on the database?
And on the other hand, its still just a property on the entity that would go lost in the rare case that a broken config is actually merged into the master. I don't see how a change like this would be easily approved for a merge.

Reply:
Any data loss is absolutely unacceptable. at least we need to be extremely aware when any changes are done. but now that a lot of properties were moved from class to config I don't have a definite overview that we didn't loose any even right now.

I am thinking whether we can at least write a unit tests that flags changes ...